compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -443,15 +443,25 @@ object Trees {
     def forwardTo: Tree[T] = fun
   }
 
+  enum ApplyKind:
+    case Regular, Using, Infix
+
   /** fun(args) */
   case class Apply[-T >: Untyped] private[ast] (fun: Tree[T], args: List[Tree[T]])(implicit @constructorOnly src: SourceFile)
     extends GenericApply[T] {
     type ThisTree[-T >: Untyped] = Apply[T]
 
-    def isUsingApply = hasAttachment(untpd.ApplyGiven)
-    def setUsingApply() = { putAttachment(untpd.ApplyGiven, ()); this }
+    def setUsingApply() =
+      putAttachment(untpd.ApplyGiven, ())
+      this
```

Comment:
A conceptual question: if `Infix` and `Using` are mutually exclusive, would it be better to use attachment for `Infix` as well to avoid creating the class `InfixApply`? Something like:

```Scala
def setApplyKind(kind: ApplyKind) = 
  putAttachment(untpd.ApplyKind, kind)
  this

def applyKind: ApplyKind =
  attachmentOrElse(untpd.ApplyKind, ApplyKind.Regular)
```

Reply:
Yes, I think that might be better. I'll try that out.